### PR TITLE
Update Helm chart metadata for recent changes

### PR DESCRIPTION
**Pull request by ZhuTopher**
This PR is made as a follow-up to the additions added in PR #13945, which added the functionality to override `livenessProbe` and `readinessProbe` in our Helm chart.

This PR just updates `Chart.yaml` and `CHANGELOG.md` to reflect these changes.

### CODE REVIEW

File: integration/kubernetes/helm-chart/alluxio/Chart.yaml
Comment (ZhuTopher) [inline integration/kubernetes/helm-chart/alluxio/Chart.yaml:21]: @jiacheliu3 @madanadit @cheyang I hope you guys have no objections?

Comment (madanadit) [reply integration/kubernetes/helm-chart/alluxio/Chart.yaml:21]: nope :)

**Review by madanadit — APPROVED** [review]

**Comment by ZhuTopher** [comment]
@jiacheliu3 I'll wait for your approval before merging ;)

**Review by apc999 — APPROVED** [review]

**Review by jiacheliu3 — APPROVED** [review]
lgtm, thanks!

**Comment by jiacheliu3** [comment]
oops there's a conflict, could you take a look? @ZhuTopher

**Comment by ZhuTopher** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**
